### PR TITLE
Enhance VMware support

### PR DESCRIPTION
**Pull request by mnowaksuse**
https://trello.com/c/O104uCeQ/178-p1-vmware-guest-install-support-in-openqa

This brings VMware via svirt support up-to-date.

Validation runs (it's enough that VM boots):
* default_install@svirt-vmware: http://nilgiri.suse.cz/tests/1951
* jeos-extratest@svirt-vmware: http://nilgiri.suse.cz/tests/1999
* jeos-main@svirt-vmware: http://nilgiri.suse.cz/tests/1998

Regression testing (it's enough that VM boots):
* jeos-extratest@svirt-xen-pv: http://nilgiri.suse.cz/tests/2003
* allpatterns@svirt-xen-hvm: http://nilgiri.suse.cz/tests/2002

Test code changes (but this can be merged independently): https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/6326.

### CODE REVIEW

**Comment by Martchus** [comment]
I'm not familiar with VMware but the code looks good in general. Let's see whether the tests are passing. Maybe change the commit message "get_cmd_output() use other connections than 'ssh' " to "Make get_cmd_output() use other connections than 'ssh'" for consistency.

File: backend/svirt.pm [hunk 1]
```diff
@@ -73,8 +73,9 @@ sub do_stop_vm {
             $self->run_cmd("$ps Remove-VM -Force -VMName $vmname");
         }
         else {
-            $self->run_cmd("virsh destroy $vmname");
-            $self->run_cmd("virsh undefine $vmname");
+            my $vmware_remote_vmm = get_var('VMWARE_REMOTE_VMM');
+            $self->run_cmd("virsh $vmware_remote_vmm destroy $vmname");
+            $self->run_cmd("virsh $vmware_remote_vmm undefine --snapshots-metadata $vmname");
```
Comment (alice-suse) [inline backend/svirt.pm]: IMHO, this $vmware_remote_vmm usage will make the svirt backend looks like supporting only hyperv and vmware. Its real default support on general linux will be blurred. So does the latter similar changes for vmware. 

What about changes 'vmware_remote_vmm' to a general name like 'vmm_connection' or whatever you prefer (no matter it is a local or remote connection to hypervisor), and just sets it to VMWARE_REMOTE_VMM when VIRSH_VMM_FAMILY is vmware?

Comment (mnowaksuse) [reply backend/svirt.pm]: Good idea @alice-suse, I changed the code according to your idea.

**Comment by mnowaksuse** [comment]
Thanks for the review @Martchus. I changed the commit message and hopefully tamed perlcritic.

File: consoles/sshVirtsh.pm [hunk 2]
```diff
@@ -330,43 +330,98 @@ sub add_interface {
 sub add_disk {
     my ($self, $args) = @_;
 
-    my $file = $self->name . $args->{dev_id} . (($self->vmm_family eq 'vmware') ? ".vmdk" : ".img");
+    my $backingfile             = $args->{backingfile};
+    my $cdrom                   = $args->{cdrom};
+    my $name                    = $self->name;
+    my $file                    = $name . $args->{dev_id} . ($self->vmm_family eq 'vmware' ? '.vmdk' : '.img');
+    my $basedir                 = '/var/lib/libvirt/images/';
```
Comment (foursixnine) [inline consoles/sshVirtsh.pm]: I wonder if it would make sense to move this to a test/worker setting... But I see no reason to block from merging...

**Comment by mnowaksuse** [comment]
codecov ain't happy, should I do something about it? (I don't even understand what's wrong.)

File: consoles/sshVirtsh.pm (hunk 2 above)
Comment (foursixnine) [reply consoles/sshVirtsh.pm]: @coolo does my comment make sense to you?

Comment (coolo) [reply consoles/sshVirtsh.pm]: to my personal taste this whole vmm_family thingy became unmaintainable long ago - there are so many specific commands in rather unexpected places. I wouldn't single out single commands. This should be rather refactored on a whole - but as long as @mnowaksuse is the only user of all of this there is no real need :(

**Pull request merged**